sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java

```diff
@@ -0,0 +1,161 @@
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/** (Internal Use Only) */
+class Utils {
+    private static final BigDecimal D_1000 = BigDecimal.valueOf(1000);
+    private static final BigDecimal D_1024 = BigDecimal.valueOf(1024);
+
+    /**
+     * Converts cpu from String to Double
+     * @param cpu cpu String
+     * @return cpu count
+     */
+    public static Double fromCpuString(String cpu) {
+        if (cpu == null) {
+            return null;
+        }
+        CpuMatcher matcher = CpuMatcher.create(cpu);
+        if (matcher.noMatch()) {
+            throw new IllegalArgumentException(String.format("Illegal cpu format : %s", cpu));
+        }
+        if (matcher.matchFraction()) {
+            return BigDecimal.valueOf(Long.parseLong(matcher.getFraction()))
+                .divide(D_1000, 1, RoundingMode.CEILING)
+                .doubleValue();
+        } else {
+            return Double.valueOf(cpu);
+        }
+    }
+
+    /**
+     * Converts memory from String to Double, in GB
+     * @param memory memory String
+     * @return memory in GB
+     */
+    public static Double fromMemoryString(String memory) {
+        if (memory == null) {
+            return null;
+        }
+        MemoryMatcher matcher = MemoryMatcher.create(memory);
+        if (matcher.noMatch()) {
+            throw new IllegalArgumentException(String.format("Illegal memory format : %s", memory));
+        }
+        if (matcher.matchGB()) {
+            return Double.valueOf(matcher.getGBString());
+        } else { // Mb
+            return BigDecimal.valueOf(Long.parseLong(matcher.getMBString()))
+                .divide(D_1024, 1, RoundingMode.CEILING)
+                .doubleValue();
+        }
+    }
+
+    /**
+     * Converts cpu count from double to String
+     * @param cpuCount cpu count, 1 core can be represented by 1 or 1000m
+     * @return cpu String
+     */
+    public static String toCpuString(double cpuCount) {
+        BigDecimal cpuDecimal = BigDecimal.valueOf(cpuCount);
+        if (isInteger(cpuCount)) {
+            return String.valueOf(cpuDecimal.intValue());
+        } else {
+            return String.format("%dm", cpuDecimal.multiply(D_1000).intValue());
+        }
+    }
+
+    /**
+     * Converts memory count from double to String
+     * @param sizeInGB memory in GB
+     * @return memory String
+     */
+    public static String toMemoryString(double sizeInGB) {
+        BigDecimal memoryDecimal = BigDecimal.valueOf(sizeInGB);
+        if (isInteger(sizeInGB)) {
+            return String.format("%dGi", memoryDecimal.intValue());
+        } else {
+            return String.format("%dMi", memoryDecimal.multiply(D_1024).intValue());
+        }
+    }
+
+    private static boolean isInteger(double sizeInGB) {
+        return (sizeInGB % 1) == 0;
+    }
+
+    private static class CpuMatcher {
```

Comment:
Please add test cases for these (package private access should be fine).

Reply:
Added.